RestaurantReservationSystem/RestaurantReservation.Db/Repositories/EmployeeRepository.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.Contexts;
+using RestaurantReservation.Db.Entities;
+using RestaurantReservation.Db.Interfaces.Repositories;
+
+namespace RestaurantReservation.Db.Repositories;
+
+public class EmployeeRepository : IEmployeeRepository
+{
+    private readonly RestaurantReservationDbContext _context;
+
+    public EmployeeRepository(RestaurantReservationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<IEnumerable<Employee>> GetAllAsync()
+    {
+        return await _context.Employees.ToListAsync();
+    }
+
+    public async Task<Employee?> GetByIdAsync(int id)
+    {
+        return await _context.Employees.FindAsync(id);
+    }
+
+    public async Task AddAsync(Employee employee)
+    {
+        await _context.Employees.AddAsync(employee);
+        await _context.SaveChangesAsync();
+    }
+
+    public async Task UpdateAsync(Employee employee)
+    {
+        _context.Employees.Update(employee);
+        await _context.SaveChangesAsync();
+    }
+
+    public async Task DeleteAsync(int id)
+    {
+        var employee = await _context.Employees.FindAsync(id);
+        if (employee != null)
+        {
+            _context.Employees.Remove(employee);
+            await _context.SaveChangesAsync();
+        }
```

Comment:
Let's prefer an early return

```suggestion
        var employee = await _context.Employees.FindAsync(id);
        if (employee is null)
        {
            return;
        }
        _context.Employees.Remove(employee);
        await _context.SaveChangesAsync();
```